### PR TITLE
Add dimensions in WFS GetFeature requests

### PR DESCRIPTION
**Pull request by arnaud-morvan**

### CODE REVIEW

**Comment by arnaud-morvan** [comment]
@fredj : Could you take a look here ?

Note that I do not understand why to not dispatch all global dimensions to each node. In my project I will have to add FLOOR dimension on approximately all layers.

**Comment by arnaud-morvan** [comment]
Note that when I put one dimension on one layer (gmf admin), it is applied on layer parent group (ol.Layer object), can't-we get this applied on all layer tree nodes, regardless of the dimensions set in admin interface ?
For me it does not make sense to put different dimensions on layers, the dimensions values should be only set on the main controller.
For WMS layers, it is passed as a GET parameter, server will use or ignore it.
For WMTS layers, if is is present in url template, server whould use it, if not it will be ignored.

File: contribs/gmf/src/directives/layertree.js [hunk 1]
```diff
@@ -290,6 +290,7 @@ gmf.LayertreeController.prototype.updateLayerDimensions_ = function(layer, node)
       var value = this.dimensions[key];
       if (value !== undefined) {
         dimensions[key] = value;
+        node.dimensions[key] = value;
       }
```
Comment (sbrunner) [inline contribs/gmf/src/directives/layertree.js]: This if looks strange for me, for me it should be like this:
```
       var value = this.dimensions[key];
       dimensions[key] = value !== undefined ? value : node.dimensions[key];
```
@fredj what do you think?

File: src/services/query.js [hunk 2]
```diff
@@ -621,6 +621,22 @@ ngeo.Query.prototype.doGetFeatureRequests_ = function(
...
+      if (dimensions) {
+        for (var key in dimensions) {
+          // get the value from the global dimensions
+          var value = this.dimensions[key];
+          if (value === undefined) {
+            // get the value from the layer default value
+            value = dimensions[key];
+          }
+          if (value !== undefined) {
+            url = goog.uri.utils.setParam(url, key, value);
```
Comment (sbrunner) [inline src/services/query.js]: Do not use `goog`!!!

File: src/services/query.js [hunk 3]
```diff
@@ -621,6 +621,22 @@ ngeo.Query.prototype.doGetFeatureRequests_ = function(
...
+      // add dimensions values
+      var dimensions = items[0].source.dimensions;
+      if (dimensions) {
+        for (var key in dimensions) {
+          // get the value from the global dimensions
+          var value = this.dimensions[key];
+          if (value === undefined) {
+            // get the value from the layer default value
+            value = dimensions[key];
+          }
```
Comment (sbrunner) [inline src/services/query.js]: Looks duplicate code from layer tree...

File: contribs/gmf/src/directives/layertree.js (hunk 1 above)
Comment (fredj) [inline contribs/gmf/src/directives/layertree.js]: `node.dimensions` are the default dimension values for a node; it should not be modified

Comment (arnaud-morvan) [inline contribs/gmf/src/directives/layertree.js]: Without this I cannot apply current dimensions on WFS GetFeature requests

File: src/services/query.js (hunk 2 above)
Comment (arnaud-morvan) [inline src/services/query.js]: Ok, can you explain, why do not use goog, what should I use instead ? Note that this comes from doGetFeatureInfoRequests_ existing code.
I see ``goog`` everywhere in this file.

Comment (sbrunner) [inline src/services/query.js]: We try to don't use goog anymore to be able to build the project without google library. 
In the master branch almost all the goog (Except goog.[asserts, requires, provide, module]) will be removes, in this case there a method in openlayers that can be used.

**Comment by arnaud-morvan** [comment]
I've changed my desktop.js file with this : 
    this.dimensions['FLOOR'] = '+0';
instead of : 
    this.dimensions = {'FLOOR': '+0'};
so I get the same global dimensions object in main controller and query service (no need to overwrite the node dimensions anymore).

I've passed the parameters directly to $http, so I do not use goog anymore.

What do you think about this updated version, is this good for merging.

**Comment by arnaud-morvan** [comment]
Note that I still have an issue on the server side, where I find really boring to add same dimensions on all layers.

**Comment by sbrunner** [comment]
For the server part you can write a script like this one: https://github.com/camptocamp/epfl_authgeoportail/blob/master/epfl_authgeoportail/scripts/manage_layers.py

**Comment by arnaud-morvan** [comment]
@fredj : Are you OK for merging this PR ?

**Pull request merged**